example-doc-generator/python/publish_docs.py

```
@@ -0,0 +1,916 @@
+    docs_repo: Path,
+    connector_name: str,
+    branch_name: str,
+    dry_run: bool,
+) -> None:
+    """Stage all changes, commit, and push the feature branch to origin."""
+    if dry_run:
+        dry(f"git add . && git commit -m 'docs: add {connector_name} connector example guide'")
+        dry(f"git push origin {branch_name}")
+        return
+    info("Committing changes...")
+    subprocess.run(["git", "add", "."], cwd=str(docs_repo), check=True)
```

Ruff (0.15.7)

[error] Starting a process with a partial executable path

(S607)